src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1435,3 +1435,45 @@ function CategoricalArrays.categorical(df::AbstractDataFrame,
     end
     DataFrame(newcols, _names(df), copycols=false)
 end
+
+"""
+    flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+
+When column `veccol` of `df` has iterable elements that define `length`, for example a `Vector` 
+of `Vector`s. Returns a DataFrame where each element of `veccol` is flattened. 
+Elements of row `i` of `df` other than `veccol` will be duplicated according to 
+the length of `df[i, veccol]`
+
+# Arguments
+
+- `df`: an `AbstractDataFrame` to flatten
+- `veccol`: a `Symbol` or `Integer` where `veccol` is a column of `df` whose 
+   elements support iteration.
+"""
+function flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+    col_to_flatten = df[!, veccol]
+    lengths = length.(col_to_flatten)
+    new_df = similar(df[!, Not(veccol)], sum(lengths))
+
+    function copy_length!(longnew, shortold, lengths)
```

Comment:
Move the definition of `flatten` and add type annotations for clarity. Also maybe call this `repeat_lengths!` or something like that?